### PR TITLE
added boilerplate.py

### PR DESCRIPTION
**Pull request by DatDerpyDude81**
Credit to @Starbuck5 for the idea.

### CODE REVIEW

**Comment by illume** [comment]
Against, because we have a full folder of examples showing this isn't the only or good way of doing things.

File: examples/boilerplate.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+screen = pygame.display.set_mode((1280, 720))
+
+clock = pygame.time.Clock()
+
+
+# main loop
+while True:
+    # fill the screen with the color purple
+    screen.fill("purple")
+# get pygame events
```
Comment (novialriptide) [inline examples/boilerplate.py:14]: Comments should be on the same indent level as the while loop

File: examples/boilerplate.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import pygame
```
Comment (novialriptide) [inline examples/boilerplate.py:1]: You should be importing `pygame` as `pg` here to keep consistency with the other examples.

**Review by novialriptide — CHANGES_REQUESTED** [review]
```python
if __name__ == "__main__":
    main()
```

You should put this at the bottom and move everything in the while loop into a `main()` function as shown here: https://github.com/pygame/pygame/blob/main/examples/sound.py

**Comment by Starbuck5** [comment]
Issue link: https://github.com/pygame/pygame/issues/3406

Thanks for making a PR.

This isn't exactly what I had in mind. I've seen in other libraries documentation "quickstart" sections that just get people what they need to get started, and that inspired me to think about the bare minimum people need to get started with pygame. One thing that is non obvious is how to set up a game loop. The challenge is how to incorporate this information into the documentation so it's consistent and easily discoverable by new users.

In my opinion, merely putting into `examples/` won't be helpful, because newbies (and the vast majority of pygame users) wouldn't think to look for it there.

**Comment by DatDerpyDude81** [comment]
I guess we could mention it in the README, 

> Issue link: #3406
> 
> Thanks for making a PR.
> 
> This isn't exactly what I had in mind. I've seen in other libraries documentation "quickstart" sections that just get people what they need to get started, and that inspired me to think about the bare minimum people need to get started with pygame. One thing that is non obvious is how to set up a game loop. The challenge is how to incorporate this information into the documentation so it's consistent and easily discoverable by new users.
> 
> In my opinion, merely putting into `examples/` won't be helpful, because newbies (and the vast majority of pygame users) wouldn't think to look for it there.

I guess we could the file in the README, or we could put this in the pygame docs website

**Review by novialriptide — CHANGES_REQUESTED** [review]

**Comment by MyreMylar** [comment]
What about adding a short 'Getting Started.rst' file under https://github.com/pygame/pygame/tree/main/docs somewhere that is then directly linked to from the main readme on github, and maybe included in the docs website as well?

.rst files work pretty well when linked to from github and can include code chunks. Could have the canonical basic 'open an empty window that won't crash' game loop in it - with a little explanation of each line and why it is needed. I'm actually surprised this doesn't exist somewhere already. Maybe it does and we just need to link it up better from the main readme?

**Pull request closed without merging**